### ISSUE TITLE
Upload of large documents fails with ConnectionError

### ISSUE THREAD

**Issue by proycon** [proycon/flat#109]
Error occurs fairly quickly however, certainly not mid-upload?

```
Traceback:

File "/scratch2/www/flat/env/lib/python3.4/site-packages/requests/packages/urllib3/connectionpool.py" in urlopen
  600.                                                   chunked=chunked)

File "/scratch2/www/flat/env/lib/python3.4/site-packages/requests/packages/urllib3/connectionpool.py" in _make_request
  356.             conn.request(method, url, **httplib_request_kw)

File "/usr/lib/python3.4/http/client.py" in request
  1125.         self._send_request(method, url, body, headers)

File "/usr/lib/python3.4/http/client.py" in _send_request
  1163.         self.endheaders(body)

File "/usr/lib/python3.4/http/client.py" in endheaders
  1121.         self._send_output(message_body)

File "/usr/lib/python3.4/http/client.py" in _send_output
  955.             self.send(message_body)

File "/usr/lib/python3.4/http/client.py" in send
  914.                 self.sock.sendall(datablock)

During handling of the above exception ([Errno 104] Connection reset by peer), another exception occurred:

File "/scratch2/www/flat/env/lib/python3.4/site-packages/requests/adapters.py" in send
  423.                     timeout=timeout

File "/scratch2/www/flat/env/lib/python3.4/site-packages/requests/packages/urllib3/connectionpool.py" in urlopen
  649.                                         _stacktrace=sys.exc_info()[2])

File "/scratch2/www/flat/env/lib/python3.4/site-packages/requests/packages/urllib3/util/retry.py" in increment
  347.                 raise six.reraise(type(error), error, _stacktrace)

File "/scratch2/www/flat/env/lib/python3.4/site-packages/requests/packages/urllib3/packages/six.py" in reraise
  685.             raise value.with_traceback(tb)

File "/scratch2/www/flat/env/lib/python3.4/site-packages/requests/packages/urllib3/connectionpool.py" in urlopen
  600.                                                   chunked=chunked)

File "/scratch2/www/flat/env/lib/python3.4/site-packages/requests/packages/urllib3/connectionpool.py" in _make_request
  356.             conn.request(method, url, **httplib_request_kw)

File "/usr/lib/python3.4/http/client.py" in request
  1125.         self._send_request(method, url, body, headers)

File "/usr/lib/python3.4/http/client.py" in _send_request
  1163.         self.endheaders(body)

File "/usr/lib/python3.4/http/client.py" in endheaders
  1121.         self._send_output(message_body)

File "/usr/lib/python3.4/http/client.py" in _send_output
  955.             self.send(message_body)

File "/usr/lib/python3.4/http/client.py" in send
  914.                 self.sock.sendall(datablock)

During handling of the above exception (('Connection aborted.', ConnectionResetError(104, 'Connection reset by peer'))), another exception occurred:

File "/scratch2/www/flat/env/lib/python3.4/site-packages/django/core/handlers/exception.py" in inner
  42.             response = get_response(request)

File "/scratch2/www/flat/env/lib/python3.4/site-packages/django/core/handlers/base.py" in _legacy_get_response
  249.             response = self._get_response(request)

File "/scratch2/www/flat/env/lib/python3.4/site-packages/django/core/handlers/base.py" in _get_response
  187.                 response = self.process_exception_by_middleware(e, request)

File "/scratch2/www/flat/env/lib/python3.4/site-packages/django/core/handlers/base.py" in _get_response
  185.                 response = wrapped_callback(request, *callback_args, **callback_kwargs)

File "/scratch2/www/flat/env/lib/python3.4/site-packages/django/contrib/auth/decorators.py" in _wrapped_view
  23.                 return view_func(request, *args, **kwargs)

File "/scratch2/www/flat/env/lib/python3.4/site-packages/flat/views.py" in upload
  518.             return upload_helper(request, namespace)

File "/scratch2/www/flat/env/lib/python3.4/site-packages/flat/views.py" in upload_helper
  495.         response = fatalerror(request,e)

File "/scratch2/www/flat/env/lib/python3.4/site-packages/flat/views.py" in fatalerror
  282.         response = render(request,'base.html', docserveerror(e))

File "/scratch2/www/flat/env/lib/python3.4/site-packages/flat/views.py" in docserveerror
  103.         raise e

File "/scratch2/www/flat/env/lib/python3.4/site-packages/flat/views.py" in upload_helper
  492.         response = flat.comm.postxml(request,"upload/" + namespace , f_folia)

File "/scratch2/www/flat/env/lib/python3.4/site-packages/flat/comm.py" in postxml
  173.     response = requests.post("http://" + settings.FOLIADOCSERVE_HOST + ":" + str(settings.FOLIADOCSERVE_PORT) + "/" + url, data=f, headers={'Content-Type':'application/xml; charset=utf-8'})

File "/scratch2/www/flat/env/lib/python3.4/site-packages/requests/api.py" in post
  110.     return request('post', url, data=data, json=json, **kwargs)

File "/scratch2/www/flat/env/lib/python3.4/site-packages/requests/api.py" in request
  56.         return session.request(method=method, url=url, **kwargs)

File "/scratch2/www/flat/env/lib/python3.4/site-packages/requests/sessions.py" in request
  488.         resp = self.send(prep, **send_kwargs)

File "/scratch2/www/flat/env/lib/python3.4/site-packages/requests/sessions.py" in send
  609.         r = adapter.send(request, **kwargs)

File "/scratch2/www/flat/env/lib/python3.4/site-packages/requests/adapters.py" in send
  473.             raise ConnectionError(err, request=request)

Exception Type: ConnectionError at /upload/
Exception Value: ('Connection aborted.', ConnectionResetError(104, 'Connection reset by peer'))
```

**Comment by proycon** [comment]
Reproduces locally in development environment as well

**Comment by proycon** [comment]
Fixed in foliadocserve 0.6.1
